### DISCUSSION TITLE
Is speed and bearing required for location features?

### DISCUSSION THREAD

**Discussion by yikun-chi** [carissalow/rapids#186] [Behavioral Features]
Hi! 

I am currenly using GPS data from a different soruce other than AWARE and only have latitude, longitude, altitude, and accuracy. Is speed / bearing actually required / used for feature calculation? The source code for Doryab and Barnett seems that they re-derive speed and bearing isn't really used. So would it be OK if I just leave the speed and bearing as NA for my dataset?  

Thanks!

**Comment by JulioV** [comment]
Hi! I believe it should be safe to set those columns as NA. Tagging @Meng6 just to confirm this is the case for Doryab location features.

**Reply by yikun-chi** [reply]
Thanks!

**Comment by Meng6** [comment]
Hi @yikun-chi , yes. @JulioV is right. For Doryab location features, it is ok if you set these columns to NAs. 
- Our code calculates `speed` based on latitude and longitude: [this function](https://github.com/carissalow/rapids/blob/master/src/features/phone_locations/doryab/add_doryab_extra_columns.py#L10).
- `bearing` is not used.

**Reply by yikun-chi** [reply]
Thanks!
